src/nm-core-utils.c

```diff
@@ -2962,7 +2962,12 @@ nm_utils_get_ipv6_interface_identifier (NMLinkType link_type,
 		memcpy (out_iid->id_u8 + 4, &addr, 4);
 		return TRUE;
 	default:
-		if (hwaddr_len == ETH_ALEN) {
+		if (hwaddr_len == G_N_ELEMENTS (out_iid->id_u8)) {
```

Comment:
this essentially detects the MAC address type, based on the length. We already do that sometimes, to distinguish ethernet vs. infiniband. I find that really ugly, because we should always explicitly know which layer2 type we have when we operate on a MAC, not detect it based on the MAC's length. The function `nm_utils_get_ipv6_interface_identifier()` already makes some decisions based on the link-type argument. Can we not instead add a `case NM_LINK_TYPE_WPAN:`?